experimental/dds/tree/src/test/HistoryEditFactory.tests.ts

```diff
@@ -50,6 +50,57 @@ describe('revert', () => {
 		expect(revertedChange.source.end.referenceSibling).to.deep.equal(secondNode.identifier);
 	});
 
+	it('handles reverting the insert of empty nodes, with multiple subsequent non-empty nodes', () => {
+		// build and insert of empty traits
+		const emptyTraitNodeId = 0 as DetachedSequenceId;
+		const emptyTraitBuild = ChangeInternal.build([], emptyTraitNodeId);
+		const emptyTraitInsert = ChangeInternal.insert(emptyTraitNodeId, {
+			referenceTrait: testTree.left.traitLocation,
+			side: Side.After,
+		});
+
+		// build and insert of non-empty traits
+		const firstDetachedId = 1 as DetachedSequenceId;
+		const firstNode = testTree.buildLeafInternal();
+		const firstBuild = ChangeInternal.build([firstNode], firstDetachedId);
+		const secondDetachedId = 2 as DetachedSequenceId;
+		const secondNode = testTree.buildLeafInternal();
+		const secondBuild = ChangeInternal.build([secondNode], secondDetachedId);
+		const insertedNodeId = 3 as DetachedSequenceId;
+		const insertedBuild = ChangeInternal.build([firstDetachedId, secondDetachedId], insertedNodeId);
```

Comment:
What is the reason for testing this somewhat complicated build scenario? What exactly is this test trying to test? Is it not sufficient to just have a single build and insert after the empty build and insert?